test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iHierarchicalIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+import org.kie.server.client.UserTaskServicesClient;
+
+public class KieServerS2iHierarchicalIntegrationTest extends AbstractCloudIntegrationTest<GenericScenario> {
+
+    protected KieServicesClient kieServicesClient;
+    protected ProcessServicesClient processServicesClient;
+    protected UserTaskServicesClient taskServicesClient;
+
+    private static final Kjar USERTASK = Kjar.USERTASK;
+    private static final String KIE_CONTAINER_DEPLOYMENT = CONTAINER_ID + "=" + USERTASK.toString();
+
+    private static final String GIT_REPOSITORY_PREFIX = "KieServerS2iHierarchicalRepository";
+
+    private static final String REPO_BRANCH = "master";
+    private static final String PROJECT_SOURCE_FOLDER = "/kjars-sources";
+    private static final String GIT_CONTEXT_DIR = "multimodule-project";
+    private static final String BUILT_KJAR_FOLDER = "usertask-project/target,signaltask-project/target";
+
+    private String repositoryName;
+
+    @Override
+    protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        repositoryName = gitProvider.createGitRepositoryWithPrefix(GIT_REPOSITORY_PREFIX, PROJECT_SOURCE_FOLDER);
+
+        DeploymentSettings kieServerS2Isettings = deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+                .withContainerDeployment(KIE_CONTAINER_DEPLOYMENT)
+                .withSourceLocation(gitProvider.getRepositoryUrl(repositoryName), REPO_BRANCH, GIT_CONTEXT_DIR, BUILT_KJAR_FOLDER)
+                .build();
+
+        return deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withKieServer(kieServerS2Isettings)
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployments().get(0));
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployments().get(0));
+        taskServicesClient = KieServerClientProvider.getTaskClient(deploymentScenario.getKieServerDeployments().get(0));
+    }
+
+    @After
+    public void deleteRepo() {
+        gitProvider.deleteGitRepository(repositoryName);
+    }
+
+    @Test
+    public void testContainerAfterExecServerS2iStart() {
+        List<KieContainerResource> containers = kieServicesClient.listContainers().getResult().getContainers();
+        assertThat(containers).isNotNull().hasSize(1);
+
+        KieContainerResource container = containers.get(0);
+        assertThat(container).isNotNull();
+        assertThat(container.getContainerId()).isNotNull().isEqualTo(CONTAINER_ID);
+
+        ReleaseId containerReleaseId = container.getResolvedReleaseId();
+        assertThat(containerReleaseId).isNotNull();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(containerReleaseId.getGroupId()).isNotNull().isEqualTo(USERTASK.getGroupId());
```

Comment:
isEqualTo's implementation calls isNotNull internally -> you can remove those calls. This comment applies across the entire PR where isNotNull is followed by another chained assetion in the same expression.

Reply:
Good point, adjusted.